### PR TITLE
add bookmarked learning resources widget

### PR DESCRIPTION
**Pull request by InsaneZein**
part of [RHCLOUD-31200](https://issues.redhat.com/browse/RHCLOUD-31200)

This is what the widget looks like in the WidgetLayout

With learning resources:
image

Without:
image

### CODE REVIEW

File: fec.config.js [hunk 1]
```diff
@@ -13,4 +13,15 @@ module.exports = {
    * Add additional webpack plugins
    */
   plugins: [],
+  moduleFederation: {
+    exposes: {
+      './RootApp': path.resolve(__dirname, './src/AppEntry.tsx'),
+      './LearningResourcesWidget': path.resolve(__dirname, './src/components/LearningResourcesWidget/LearningResourcesWdiget.tsx'),
```
Comment (fhlavac) [inline fec.config.js:19]: We probably don't want to expose the widget until we agree on the common approach - https://redhat-internal.slack.com/archives/C05M0SNTLM8/p1710170397998599

File: src/components/LearningResourcesWidget/LearningResourcesWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+    <Link onClick={() => updateDocumentTitle(title)} to={pathname}>
+      {title}
+    </Link>
+  );
+};
+
+const LearningResourcesWidget: React.FunctionComponent = () => {
+  const { quickStarts, bookmarks } = useQuickStarts();
+  console.log(bookmarks);
```
Comment (fhlavac) [inline src/components/LearningResourcesWidget/LearningResourcesWidget.tsx:41]:
```suggestion
```

File: src/components/LearningResourcesWidget/LearningResourcesWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                  pathname={spec.link?.href || ''}
+                />
+              </TextContent>
+              <Flex direction={{ default: 'row' }}>
+                <FlexItem
+                  style={{ marginRight: 'var(--pf-v5-global--spacer--sm)' }}
+                >
+                  {spec.type && (
+                    <Label
+                      className="pfext-quick-start-tile-header--margin"
```
Comment (fhlavac) [inline src/components/LearningResourcesWidget/LearningResourcesWidget.tsx:63]: does this have any effect?

File: src/components/LearningResourcesWidget/LearningResourcesWidget.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.widget-learning resources {
+  column-width: 300px;
```
Comment (fhlavac) [inline src/components/LearningResourcesWidget/LearningResourcesWidget.scss:2]: Not sure where this is being used, but do we want the pixel value here? Something relative might work better

**Comment by fhlavac** [comment]
@InsaneZein looks good! Just some small comments

**Review by epwinchell — APPROVED** [review]

**Pull request merged**